src/lib.rs

```diff
@@ -91,39 +88,50 @@ impl<Manager> Clone for Client<Manager> {
 struct Inner<Manager> {
     _manager: Manager,
     callbacks: Mutex<Callbacks>,
-    networking_sockets_data: Mutex<NetworkingSocketsData<Manager>>,
+    lib: Arc<sys::steam_api>,
+    // networking_sockets_data: Mutex<NetworkingSocketsData<Manager>>,
 }
 
 struct Callbacks {
     callbacks: HashMap<i32, Box<dyn FnMut(*mut c_void) + Send + 'static>>,
     call_results: HashMap<sys::SteamAPICall_t, Box<dyn FnOnce(*mut c_void, bool) + Send + 'static>>,
 }
 
-struct NetworkingSocketsData<Manager> {
-    sockets: HashMap<
-        sys::HSteamListenSocket,
-        (
-            Weak<networking_sockets::InnerSocket<Manager>>,
-            Sender<networking_types::ListenSocketEvent<Manager>>,
-        ),
-    >,
-    /// Connections to a remote listening port
-    independent_connections: HashMap<sys::HSteamNetConnection, Sender<()>>,
-    connection_callback: Weak<CallbackHandle<Manager>>,
-}
+// struct NetworkingSocketsData<Manager> {
+//     sockets: HashMap<
+//         sys::HSteamListenSocket,
+//         (
+//             Weak<networking_sockets::InnerSocket<Manager>>,
+//             Sender<networking_types::ListenSocketEvent<Manager>>,
+//         ),
+//     >,
+//     /// Connections to a remote listening port
+//     independent_connections: HashMap<sys::HSteamNetConnection, Sender<()>>,
+//     connection_callback: Weak<CallbackHandle<Manager>>,
+// }
 
 unsafe impl<Manager: Send + Sync> Send for Inner<Manager> {}
 unsafe impl<Manager: Send + Sync> Sync for Inner<Manager> {}
 unsafe impl<Manager: Send + Sync> Send for Client<Manager> {}
 unsafe impl<Manager: Send + Sync> Sync for Client<Manager> {}
 
-/// Returns true if the app wasn't launched through steam and
-/// begins relaunching it, the app should exit as soon as possible.
+// /// Returns true if the app wasn't launched through steam and
+// /// begins relaunching it, the app should exit as soon as possible.
+// ///
+// /// Returns false if the app was either launched through steam
+// /// or has a `steam_appid.txt`
+// pub fn restart_app_if_necessary(app_id: AppId) -> bool {
+//     unsafe { sys::SteamAPI_RestartAppIfNecessary(app_id.0) }
+// }
+
+/// Path to the steamworks sdk library.
 ///
-/// Returns false if the app was either launched through steam
-/// or has a `steam_appid.txt`
-pub fn restart_app_if_necessary(app_id: AppId) -> bool {
-    unsafe { sys::SteamAPI_RestartAppIfNecessary(app_id.0) }
+/// Steamworks sdk library is expected to exist next to the binary.
+pub fn steam_library_path() -> std::path::PathBuf {
+    let lib_name = libloading::library_filename("steam_api");
+
+    let exe_path = std::env::current_exe().unwrap();
+    exe_path.parent().unwrap().join(lib_name)
```

Comment:
Wasn't this the default on Windows? Or did it look relative to `PATH`/`cwd`? Or all of them?

https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryw#remarks :see_no_evil:

Reply:
I thought so too and I might still be doing something wrong but when I passed the `library_filename` to `libloading::Library::new(path)` it didn't work. Specifying the full path to the exe did work so 🤷.